inference-engine/src/transformations/include/transformations/convert_reduce_to_pooling.hpp

```diff
@@ -227,21 +227,46 @@ ngraph::matcher_pass_callback convert_reduce_to_pooling() {
             input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "/pool");
             new_ops.push_back(input.get_node_shared_ptr());
         } else if (std::is_same<T, ngraph::opset1::ReduceSum>()) {
-            input = std::make_shared<ngraph::opset1::AvgPool>(input,
-                                                              strides,
-                                                              pads_begin,
-                                                              pads_end,
-                                                              kernel,
-                                                              true,
-                                                              ngraph::op::RoundingType::FLOOR);
-
-            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "/pool");
-            new_ops.push_back(input.get_node_shared_ptr());
-
-            input = std::make_shared<ngraph::opset1::Multiply>(input,
-                    ngraph::opset1::Constant::create(reduce->input(0).get_element_type(), ngraph::Shape{1}, {reduction_dims_count}));
-            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "/mul");
-            new_ops.push_back(input.get_node_shared_ptr());
+            if (input.get_element_type().is_integral()) {
+                auto orig_type = input.get_element_type();
+                auto real_type = ngraph::element::f32;
+
+                input = std::make_shared<ngraph::opset1::Convert>(input, real_type);
```

Comment:
Look like the part with AvgPool creation is common for if-else statement. So I suggest to move it to lamda instead of copying this code.

Reply:
Lambda is not so nice here. Can only suggest the solution with convert layers hidden under `if` conditions.